pkg/cluster/internal/providers/docker/provision.go

```diff
@@ -131,6 +132,7 @@ func createContainer(args []string) error {
 	if err := exec.Command("docker", args...).Run(); err != nil {
 		return errors.Wrap(err, "docker run error")
 	}
+	time.Sleep(10 * time.Second) // debug
```

Comment:
The “ ERROR: this script needs /sys/fs/cgroup/cgroup.procs to be empty” issue was happening when `docker exec -i` was executed before systemd initializes `/sys fs/cgroup`.

So this sleep works as a workaround.

Reply:
do we have some way to "detect it" instead of sleeping?

Reply:
I guess we could start watching for a log line in the node output if this is the case.

Reply:
I'm not sure.

Grepping `Reached target .*Multi-User System.*` is fine?

Reply:
I guess. It's unfortunate that this is the case, we have codepaths that should be fine without a running system (e.g. catting the version file, building kubeadm config), and speed is definitely a priority even if we haven't improved on it much lately.

Maybe we can conditionally wait on this when rootless?

Reply:
Opened https://github.com/kubernetes-sigs/kind/pull/2421

> Maybe we can conditionally wait on this when rootless?

I don't think this is specific to rootless, though specific to cgroup v2.